feat(d2k): Add Tutorial Mission Mode

YibinLong opened the pull request:
## Summary
This PR adds a comprehensive Tutorial Mission Mode to Dune 2000 (D2K) to help new players learn the game mechanics.

## Features
- **14 Progressive Tutorial Objectives** covering:
  - Camera controls (scrolling, home key)
  - Unit selection (click, drag-select)
  - Unit movement
  - Control groups (Ctrl+number)
  - MCV deployment
  - Concrete foundation placement
  - Power generation (Wind Trap)
  - Resource harvesting (Refinery + Harvester)
  - Infantry production (Barracks + Light Infantry)
  - Vehicle production (Light Factory + Trikes)
  - Combat basics vs Harkonnen forces

- **Tutorial Button** added to the D2K singleplayer menu with save/resume support
- **Full Localization Support** via Fluent translation files
- **Custom Tutorial Map** with player starting area and enemy Harkonnen base

## Files Changed
- `OpenRA.Mods.Common/Widgets/Logic/MainMenuLogic.cs` - Tutorial button logic
- `mods/d2k/chrome/mainmenu.yaml` - UI button definition
- `mods/d2k/fluent/chrome.ftl` - Menu translations
- `mods/d2k/fluent/tutorial.ftl` - Tutorial message translations
- `mods/d2k/maps/tutorial/*` - Tutorial map assets and Lua script
- `mods/d2k/missions.yaml` - Mission registration
- `mods/d2k/mod.yaml` - Mod configuration

## Testing
- [x] Tutorial objectives progress correctly in sequence
- [x] Tutorial button appears in singleplayer menu
- [x] Save/resume functionality works
- [x] All tutorial messages display correctly

YibinLong committed:
feat(d2k): add tutorial map and mission assets

- Add tutorial map with Atreides vs Harkonnen scenario
- Add Lua script with 14 progressive objectives teaching:
  - Camera movement and navigation
  - Unit selection and movement
  - Control groups
  - MCV deployment
  - Building construction (power, refinery, barracks, factory)
  - Resource harvesting
  - Unit training and combat
- Add tutorial fluent localization strings

YibinLong committed:
feat: implement tutorial button logic with save/resume support

- Add tutorial button click handler that finds tutorial map by title
- Add FindTutorialSave() to locate existing tutorial saves
- Add StartNewTutorial() and LoadTutorialSave() methods
- Show resume/start new dialog when tutorial save exists
- Add FluentReference constants for dialog strings

YibinLong committed:
feat(d2k): add Tutorial button to singleplayer menu

- Add TUTORIAL_BUTTON between Missions and Load buttons
- Adjust singleplayer menu height and button positions
- Add localization for Tutorial button text
- Add dialog strings for tutorial resume/start new prompt
- Register tutorial in missions.yaml

YibinLong committed:
feat(d2k): add tutorial mission mode

- Add tutorial map with player starting units (MCV, 3 Light Infantry)
- Add Harkonnen enemy base with passive defense force
- Implement 15-step tutorial objectives covering:
  - Camera movement and navigation
  - Unit selection (click and drag)
  - Unit movement and control groups
  - MCV deployment to Construction Yard
  - Building placement (concrete, wind trap, refinery, barracks)
  - Resource harvesting mechanics
  - Infantry and vehicle production
  - Basic combat against enemy forces
- Add Tutorial button to main menu with resume/restart dialog
- Add localized tutorial messages in Fluent format
- Enemy units remain passive until attacked, then fight back

JovialFeline commented on mods/d2k/maps/tutorial/tutorial.lua:
```diff
@@ -0,0 +1,523 @@
+
+Check6_PlaceConcrete = function()
+	-- Time-based completion handled in objective function
+	-- (Concrete actors self-destruct and become terrain, so we can't count them)
```
While Map.TerrainType _is_ capable of counting concrete cells near the Construction Yard, it would be simpler to use `OnBuildingPlaced(Atreides, function(_, actor)` and check the `concretea` type.

JovialFeline commented on mods/d2k/maps/tutorial/tutorial.lua:
```diff
@@ -0,0 +1,523 @@
+		UserInterface.SetMissionText(UserInterface.GetFluentMessage("tutorial-mission-text-move"))
+	end)
+
+	-- Auto-complete after time since movement detection is complex
```
Do you feel ordering them to another (unbuildable) cell would be effective? It's how RA2 does it, at least.

PunkPun reviewed (commented):
👋 Hi

Thanks for your contribution

Personally, I'm not a huge fan of text-based tutorials. It also seems a bit redundant as the first D2K mission is already a tutorial, though it is less handholdy and assumes you will figure things out to achieve the given task. As a designer this is the type of tutorials I enjoy most, but sure, the current implementation is less than perfect

YibinLong committed:
fix: fix empty array usage and make tutorial references optional

Replaces the use of the C# 12 array literal with Array.Empty<byte>() for compatibility in Map.cs. Marks tutorial dialog string references as optional in MainMenuLogic.cs to prevent errors if they are missing.

YibinLong commented:
Hey, thanks for your review.

That makes sense, I like tutorials that are a bit more handholdy because I think it makes the game more accessible for newer players, but I see how the first mission is similar to the tutorial as well. If I have time I'll look into improving the implementation of the existing tutorials!

YibinLong committed:
fix: add tutorial resume dialog and update D2K message references

Introduces a 'Resume Tutorial' dialog to the chrome.ftl files for cnc, ra, and ts mods. Updates the D2K tutorial map to include campaign Fluent messages and removes a duplicate tutorial Fluent message reference from d2k/mod.yaml. Also fixes a C# syntax issue in Map.cs for empty byte array initialization.
